src/include/miopen/median/problem_description.hpp

```diff
@@ -0,0 +1,250 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/tensor.hpp>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace median {
+
+struct FwdProblemDescription : public ProblemDescriptionBase
+{
+    FwdProblemDescription(const TensorDescriptor& inputDesc_,
+                          const TensorDescriptor& outputDesc_,
+                          const TensorDescriptor& indicesDesc_,
+                          const uint64_t dim_,
+                          const bool keepdim_)
+        : inputDesc(inputDesc_),
+          outputDesc(outputDesc_),
+          indicesDesc(indicesDesc_),
+          dim(dim_),
+          keepdim(keepdim_)
+    {
+        IsValidNumDims();
+        IsValidDim();
+        IsRightLength();
+        IsSameType();
+    }
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+    const TensorDescriptor& GetIndicesDesc() const { return indicesDesc; }
+    uint64_t GetDim() const { return dim; }
+    bool GetKeepDim() const { return keepdim; }
+
+    bool IsValidNumDims() const
+    {
+        if(inputDesc.GetNumDims() > 5)
+        {
+            MIOPEN_THROW(
+                miopenStatusBadParm,
+                "MedianForward: A tensor with dimensions greater than 5 is not supported yet.");
+        }
+
+        return true;
+    }
```

Comment:
Logically, the problem is valid even this check fail. This check is only required for some specific solvers (1 solver in this context). Because of that, it must be called in `IsApplicable` without throwing exception on check fail.

Reply:
solved